### PR TITLE
Refactors config files

### PR DESCRIPTION
**Pull request by edublancas**
I refactored the logic to store and retrieve the data from the config files. This is a pre-requisite to start working on the improved onboarding workflow where we have a "tutorial" to point to the user what do to next since we need to add a few extra flags to the config files to track the state.

The tests remain unchanged (I only modified a few and deleted some that are outdated)

### CODE REVIEW

**Comment by edublancas** [comment]
can you review this @idomic ?

File: src/ploomber/config.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import warnings
+import abc
+from collections.abc import Mapping
+
+import yaml
+
+
+class Config(abc.ABC):
+    """And abstract class to create configuration files (stored as YAML)
```
Comment (idomic) [inline src/ploomber/config.py:9]: Typo

File: src/ploomber/telemetry/telemetry.py [hunk 2]
```diff
@@ -57,6 +55,35 @@
...
+    version_check_enabled: bool = True
+    cloud_key: str = None
+    stats_enabled: bool = True
+
+    @classmethod
+    def path(cls):
+        return Path(check_dir_exist(CONF_DIR), DEFAULT_USER_CONF)
+
+
+class Internal(Config):
```
Comment (idomic) [inline src/ploomber/telemetry/telemetry.py:70]: Maybe call it ploomber settings?

File: tests/telemetry/test_telemetry.py [hunk 3]
```diff
@@ -102,24 +144,18 @@ def test_env_var_takes_precedence(monkeypatch,
...
 
-    stats = Path('stats')
-    stats.mkdir()
-
-    # This isn't a first time usage since the config file doesn't exist yet.
-    assert not telemetry.check_first_time_usage()
-    (stats / 'config.yaml').write_text("stats_enabled: True")
-
     assert telemetry.check_first_time_usage()
+    assert not telemetry.check_first_time_usage()
```
Comment (idomic) [inline tests/telemetry/test_telemetry.py:159]: How does this pass, now that we don't set the conf?

**Review by idomic — CHANGES_REQUESTED** [review]
Looks good besides comments

File: src/ploomber/config.py (hunk 1 above)
Comment (edublancas) [reply src/ploomber/config.py:9]: fixed

File: src/ploomber/telemetry/telemetry.py (hunk 2 above)
Comment (edublancas) [reply src/ploomber/telemetry/telemetry.py:70]: I want to be more explicit that these are for internal use and others are for users.

File: tests/telemetry/test_telemetry.py (hunk 3 above)
Comment (edublancas) [reply tests/telemetry/test_telemetry.py:159]: I added a one-off flag to the config to replicate this functionality

Comment (idomic) [reply tests/telemetry/test_telemetry.py:159]: Cool just make sure it works via the dashboard

**Review by idomic — APPROVED** [review]
Just make sure to test/squash commits.
Besides, lgtm

**Pull request merged**